### DISCUSSION TITLE
[Bug] Discount calculations are wrong

### DISCUSSION THREAD

**Discussion by mewa** [swellstores/community#436] [Help]
According to the [docs](https://www.swell.is/help/marketing/discounts/discount-considerations#discount-considerations) discounts are calculated independently.
![2023-11-08-143022_736x493_scrot](https://github.com/swellstores/community/assets/7347194/f07eaca5-a107-4018-89b6-15b4568dfc4d)

But if you create e.g. a Buy X get Y discount with the following structure:
* buy 2 of X, get -10% off 1 quantity of X
* buy 3 of X, get -30% off 1 quantity of X

Let's assume product A from category X costs 20 EUR.

If you add 2x of product A, the applied discount is correct, i.e. 2 EUR. However, if you add 3x of product A, the applied discount is 7.40 EUR.

I would've expected to see 8 EUR. Or even 6 EUR if the latter discount (somehow) cancelled the first one (on a side note, such behaviour would actually be useful to define bulk pricing ladders).

Edit: The difference seems to be 30% of the already discounted price (i.e. 20 - 2 = 18 EUR).

**Comment by logeshswell** [comment]
The expected discount value is 8 EUR but it would be very helpful if you share the cart link.

**Reply by logeshswell** [reply]
I doubt that the discount might have been applied based on the minimum and maximum quantity conditions present in the coupon code.
